Apollo portal playbook

neoformit opened the pull request:
Required the following in ansible/playbooks/group_vars/newportalvms/vault:

```
admin_email:           <str>
admin_password:        <str>
user_support_email:    <str>
postgres_db_password:  <str>
django_secret_key:     <str>
smtp_password:         <str>
slack_error_api_key:   <str>
recaptcha_public_key:  <str>
recaptcha_private_key: <str>
```

jlqfab commented on ansible/playbooks/group_vars/newportalvms/vars:
```diff
@@ -0,0 +1,83 @@
+  password: "{{ admin_password }}"
+
+smtp:
+  hostname: smtp.office365.com
```
This needs to use the `mail.usegalaxy.org.au` postal server as per the current Drupal apollo-portal

neoformit replied:
Yeah I'm working on getting the Postal server to work - I can make a separate PR for this.

jlqfab commented on ansible/playbooks/hosts:
```diff
@@ -47,11 +47,11 @@ apollo-monitor.genome.edu.au
-
+[dev_portal_webservers]
+ubuntu22-test.genome.edu.au  ansible_connection=ssh  ansible_user=ubuntu
 
```
this should go in it's own inventory file. See `ansible/playbooks/buildmonitor.inventory` for example.

jlqfab replied:
also it should be defined as `newportalvms` to match the group_vars.

jlqfab commented on ansible/playbooks/playbook-dev-deploy-apollo-portal.yml:
```diff
@@ -0,0 +1,15 @@
+---
+- hosts: dev_portal_webservers
```
this should be `newportalvms` to match the group_vars.

jlqfab commented on ansible/playbooks/playbook-dev-deploy-apollo-portal.yml:
```diff
@@ -0,0 +1,15 @@
+  become_method: sudo
+  connection: "ssh"
+
+  vars_files:
```
not needed if define the VM as a member of `newportalvms` in the inventory

jlqfab commented on ansible/roles/geerlingguy.ansible/.github/workflows/release.yml:
Is this applicable to this project or this this galaxy specific??? Same goes for the other `geerlinguy.ansible/.github` files - are they needed?

neoformit replied:
I don't know - `galaxy-ansible` installed them. So we should either commit, or gitignore them?

jlqfab commented on ansible/roles/geerlingguy.ansible/README.md:
why is this here? the `geerlingguy.ansible` role is installed via the `ansible-galaxy/requirements.yml` on the deployment server. It is not required on wanted on the web server.

neoformit replied:
Then we should add this to the gitignore?

jlqfab commented on ansible/roles/geerlingguy.ansible/.ansible-lint:
I think the geerlingguy.ansible role should be removed - it isn't (or shouldn't be) part of the portal deployment. The geerlingguy.ansible role is installed via the ansible-galaxy/requirements.yml on the deployment server. It is not required on wanted on the web server.

jlqfab commented on ansible/roles/geerlingguy.git/.ansible-lint:
the geerlingguy.git role should be installed via ansible-galaxy if required, not included in the apollo infrastructure code base

jlqfab commented on ansible/roles/geerlingguy.postgresql/README.md:
As with the other geerlingguy ansible roles - these should be installed via ansible-galaxy, _not_ included in the apollo infrastructure code base.

jlqfab reviewed (changes requested):
As it stands. This can't be merged. It includes geerlingguy roles into the codebase, when they only need to be added to the `ansible-galaxy/requirements.yml` file. Also, smtp settings need to be corrected (see comments and as per conversations), and a separate inventory file should be created that adds the VM to the `newportalvms` group, which in turn provides access to those group_vars.

neoformit commented:
I have reset the last commit (the portal role/playbook) and added the following to `.gitignore`:

```
# Development roles
ansible/roles/geerlingguy.ansible/
ansible/roles/geerlingguy.git/
```

Then committed the changes again. Let me know if you're happy with this and I will force-push (this will wipe those roles from the repo history). I have also fixed the inventory files and host name.

jlqfab commented:
The  ansible-galaxy roles from geerlingguy should be installed in your `$HOME/.ansible/roles` directory when you install with ansible-galaxy. So they shouldn't be turning up under the code repo.

jlqfab commented on ansible/playbooks/hosts:
```diff
@@ -47,12 +47,9 @@ apollo-monitor.genome.edu.au
 apollo-portal.genome.edu.au
 #apollo-portal-main.genome.edu.au # use this when apollo-portal redirects to apollo-portal-offline
 apollo-deploy.genome.edu.au
-ubuntu22-test.genome.edu.au
```
can you put ubuntu22-test.genome.edu.au back in the hosts inventory file. ie just revert ansible/playbooks/hosts  to what it was before you made any changes (ie it shouldn't be included in this PR). otherwise I need to add it back for system updates.

jlqfab reviewed (approved):
Thanks. Looks good!
